### ISSUE TITLE
[MNT]: wheel of 3.5.0 apears to depend on setuptools-scm which apears to be unintentional

### ISSUE THREAD

**Issue by franzhaas** [matplotlib/matplotlib#21783]
### Summary

matplotlib does pull in setuptools as a runtime dependency.

if setuptools is installed it uses the pkg_resources to look for resources.

Is this on purpose? The code in the setup.py apears to me to mean that the dependency should not be a dependency for production packages, or am i wrong?

I would like to get rid of this dependency.

### Proposed fix

If possible the nicest solution would be to just drop the dependency.

I am unsure if there is really a use case for setuptools_scm at runtime (as opposed to setup time).

Thansk for the great work!

**Comment by epeisach** [comment]
I second this.  setuptools-scm being installed on your system will affect other packages being built.  The check that .git directory exists to decide on the pull in as an install requirement is wrong. 

Either setup.py needs to be changed - our when building packages in your CI - you must remove the .git directory.

**Comment by tacaswell** [comment]
I am sold on the wheels not depending on setuptools-scm as a matter of principle, but if it being installed or not is breaking other packages for you @epeisach my knee-jerk reaction is that there are bugs in the other packages!

**Comment by dopplershift** [comment]
setuptools_scm should only ever be a dependency of building the wheels, not runtime. @tacaswell I think this is a little suspect:

https://github.com/matplotlib/matplotlib/blob/d536acffaf405dba137233fde70bc81bd294327f/setup.py#L315-L328

Do you remember why that `install_requires` portion was added?

**Comment by dopplershift** [comment]
In MetPy I chose to go with:
```python
    try:
        from setuptools_scm import get_version
        return get_version(root='../..', relative_to=__file__,
                           version_scheme='post-release')
    except (ImportError, LookupError):
        try:
            from importlib.metadata import PackageNotFoundError, version
        except ImportError:  # Can remove when we require Python > 3.7
            from importlib_metadata import PackageNotFoundError, version

        try:
            return version(__package__)
        except PackageNotFoundError:
            return 'Unknown'
```
and avoid polluting `install_requires`. I've also moved as much as possible to put metadata in `setup.cfg`, so I don't even have the option for this IMO anti-pattern.

**Comment by QuLogic** [comment]
I suggest we do that for 3.6.0, and do the minimal patching in the above PR for 3.5.1.

**Comment by pcorpet** [comment]
I still have setuptools-scp as a dep of matplotlib 3.8.0 in my lock file. Can you help me debug please? I see that it's in the `requires_dist` on PyPi:  https://pypi.org/pypi/matplotlib/3.8.0/json. Aren't you pushing the lib to PyPi as a wheel?

**Comment by ksunden** [comment]
@pcorpet see #26932

TLDR is that setuptools changed their sdist metadata generation, so while the wheels don't report that dependency the sdist (and thus pypi) report it. Will be fixed for 3.8.1
